rewrite-maven/src/main/java/org/openrewrite/maven/IdentifyUnreachableRepos.java

```diff
@@ -0,0 +1,155 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import lombok.AllArgsConstructor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.HttpSenderExecutionContextView;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.ipc.http.HttpSender;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.*;
+
+public class IdentifyUnreachableRepos extends Recipe {
+    private static final XPathMatcher REPOSITORY_URL_MATCHER = new XPathMatcher("/project/repositories/repository/url");
+    private static final XPathMatcher PLUGIN_REPOSITORY_URL_MATCHER = new XPathMatcher("/project/pluginRepositories/pluginRepository/url");
+    private static final XPathMatcher DISTRIBUTION_MANAGEMENT_REPOSITORY_URL_MATCHER = new XPathMatcher("/project/distributionManagement/repository/url");
+    private static final XPathMatcher DISTRIBUTION_MANAGEMENT_SNAPSHOT_REPOSITORY_URL_MATCHER = new XPathMatcher("/project/distributionManagement/snapshotRepository/url");
+
+    public static final Map<String, String> KNOWN_DEFUNCT;
+    public static final Set<String> KNOWN_ACTIVE;
+
+    static {
+        // KNOWN_DEFUNCT is a growing list containing repositories that are known to be defunct
+        Map<String, String> knownDefunct = new HashMap<>();
+        knownDefunct.put("maven.glassfish.org/content/groups/public", "maven.java.net");
+        knownDefunct.put("abcmaven.com", null);
+        KNOWN_DEFUNCT = Collections.unmodifiableMap(knownDefunct);
+
+        //KNOWN_ACTIVE is a (growing) list containing repositories that are known to always be active
+        Set<String> knownNonDefunct = new HashSet<>();
+        KNOWN_ACTIVE = Collections.unmodifiableSet(knownNonDefunct);
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Identify potentially dead repositories";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Identify remote repositories that are not responding to an HTTP/HTTPS request.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new UnreachableRepoMarkerVisitor();
+    }
+
+    private static class UnreachableRepoMarkerVisitor extends MavenIsoVisitor<ExecutionContext> {
+        @Override
+        public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
+            // if the tag is <...repository>, accordingly mark the repo URL if it is unreachable
+            if (isRepositoryUrlTag()) {
+                // get the url subtag
+                String url = tag.getValue().orElse(null);
+                if (url != null) {
+                    if (url.startsWith("$")) {
```

Comment:
It is not guaranteed that the entire URL would be a property. 
Could easily have a situation like "https://repo.whatever.org/${subdomain}"
I don't believe that situation is _common_ but it is possible.